plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -183,10 +192,60 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
         }
       }
     }
-
+    if (layer is SymbolLayer) {
+      // Only apply cluster for SymbolManager
+      initClusterLayers()
+    }
     updateSource()
   }
 
+  private fun initClusterLayers() {
+    annotationConfig?.annotationSourceOptions?.clusterOptions?.let {
+      it.colorLevels.forEachIndexed { level, _ ->
+        val clusterLevelLayer = createClusterLevelLayer(level, it.colorLevels)
+        if (!style.styleLayerExists(clusterLevelLayer.layerId)) {
+          style.addLayer(clusterLevelLayer)
+        }
+      }
+      val clusterTextLayer = createClusterTextLayer()
+      if (!style.styleLayerExists(clusterTextLayer.layerId)) {
+        style.addLayer(clusterTextLayer)
+      }
+    }
+  }
+
+  private fun createClusterLevelLayer(level: Int, colorLevels: List<Pair<Int, Int>>): CircleLayer {
```

Comment:
Could simplify here to `private fun createClusterLevelLayer(level: Int, colorLevels: List<Pair<Int, Int>>) = circleLayer() { ... }`